images/pr-creator/Containerfile

```diff
@@ -1,26 +1,20 @@
-FROM quay.io/kubevirtci/golang:v20240814-9b6c4af
+FROM quay.io/kubevirtci/golang:v20251022-6eb3ab1
 
 ENV GIMME_GO_VERSION=1.22.6
 
 RUN set -x && \
-    git clone https://github.com/kubernetes/test-infra.git && \
+    git clone --depth 1 --revision e4d87dae72dbf1754625cbf112afc535307c1db4 https://github.com/kubernetes/test-infra.git && \
     cd ./test-infra && \
-    git checkout e4d87dae72dbf1754625cbf112afc535307c1db4 && \
-    /usr/local/bin/runner.sh /bin/sh -ce 'go build -o ./robots/pr-creator/pr-creator ./robots/pr-creator/main.go' && \
-    mv ./robots/pr-creator/pr-creator /usr/local/bin/ && \
+    /usr/local/bin/runner.sh /bin/sh -ce 'go mod vendor && go build -o /usr/local/bin/pr-creator ./robots/pr-creator && go clean -cache -modcache' && \
     chmod +x /usr/local/bin/pr-creator && \
-    /usr/local/bin/runner.sh /bin/sh -ce 'go clean -cache -modcache' && \
     cd .. && rm -rf ./test-infra
 
 RUN set -x && \
-    git clone https://github.com/kubevirt/project-infra.git && \
-    cd ./project-infra && \
-    cp ./hack/git-askpass.sh /usr/local/bin && \
-    cp ./hack/git-pr.sh /usr/local/bin && \
+    cd /project-infra && \
+    cp ./hack/git-askpass.sh ./hack/git-pr.sh /usr/local/bin && \
     chmod +x /usr/local/bin/git-askpass.sh /usr/local/bin/git-pr.sh && \
-    /usr/local/bin/runner.sh /bin/sh -ce 'go build -o /usr/local/bin/labels-checker ./robots/cmd/labels-checker' && \
-    chmod +x /usr/local/bin/labels-checker && \
-    cd .. && rm -rf ./project-infra
+    /usr/local/bin/runner.sh /bin/sh -ce 'go build -o /usr/local/bin/labels-checker ./robots/cmd/labels-checker && go clean -cache -modcache' && \
+    chmod +x /usr/local/bin/labels-checker
```

Comment:
It might make sense to update this to be a multistage image build similar to what we do with the prow-deploy image.